eras/babbage/test-suite/src/Test/Cardano/Ledger/Babbage/Serialisation/Generators.hs

```diff
@@ -0,0 +1,348 @@
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans -fno-warn-unused-imports #-}
+
+module Test.Cardano.Ledger.Babbage.Serialisation.Generators where
+
+import Cardano.Ledger.Alonzo.Data (AuxiliaryData (..), BinaryData, Data (..), dataToBinaryData)
+import Cardano.Ledger.Alonzo.Language
+import Cardano.Ledger.Alonzo.Rules.Utxo (UtxoPredicateFailure (..))
+import Cardano.Ledger.Alonzo.Rules.Utxos (TagMismatchDescription (..), UtxosPredicateFailure (..))
+import Cardano.Ledger.Alonzo.Rules.Utxow (AlonzoPredFail (..))
+import Cardano.Ledger.Alonzo.Scripts
+  ( CostModel (..),
+    ExUnits (..),
+    Prices (..),
+    Script (..),
+    Tag (..),
+  )
+import Cardano.Ledger.Alonzo.TxInfo (FailureDescription (..), ScriptResult (..))
+import Cardano.Ledger.Alonzo.TxWitness
+import Cardano.Ledger.Babbage (BabbageEra)
+import Cardano.Ledger.Babbage.PParams
+import Cardano.Ledger.Babbage.Tx
+import Cardano.Ledger.Babbage.TxBody
+  ( BabbageBody,
+    Datum (..),
+    TxBody (..),
+    TxOut (..),
+  )
+import Cardano.Ledger.CompactAddress (compactAddr)
+import Cardano.Ledger.Compactible (toCompact)
+import qualified Cardano.Ledger.Core as Core
+import Cardano.Ledger.Era (Crypto, Era, ValidateScript (..))
+import Cardano.Ledger.Hashes (ScriptHash)
+import Cardano.Ledger.Shelley.Constraints (UsesScript, UsesValue)
+import Control.Applicative (liftA2)
+import Data.Int (Int64)
+import Data.Map (Map)
+import qualified Data.Map as Map
+import Data.Maybe (fromJust, fromMaybe)
+import qualified Data.Set as Set
+import Data.Text (pack)
+import Numeric.Natural (Natural)
+import Plutus.V1.Ledger.Api (defaultCostModelParams)
+import qualified PlutusTx as Plutus
+import Test.Cardano.Ledger.Alonzo.Scripts (alwaysFails, alwaysSucceeds)
+import Test.Cardano.Ledger.Shelley.ConcreteCryptoTypes (Mock)
+import Test.Cardano.Ledger.Shelley.Serialisation.EraIndepGenerators ()
+import Test.Cardano.Ledger.ShelleyMA.Serialisation.Generators (genMintValues)
+import Test.QuickCheck
+
+instance
+  ( Era era,
+    UsesValue era,
+    Mock (Crypto era),
+    Arbitrary (Core.Value era),
+    Arbitrary (Core.Script era)
+  ) =>
+  Arbitrary (TxOut era)
+  where
+  arbitrary =
+    TxOut
+      <$> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+
+instance
+  ( UsesValue era,
+    Mock (Crypto era),
+    BabbageBody (era),
+    Arbitrary (Core.Value era),
+    Arbitrary (Core.Script era)
+  ) =>
+  Arbitrary (TxBody era)
+  where
+  arbitrary =
+    TxBody
+      <$> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> genMintValues
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+
+instance Arbitrary (Data era) where
```

Comment:
I think we should be able to re-use a lot of the alonzo generators, such at the Data one.

https://github.com/input-output-hk/cardano-ledger/blob/7aae0d97b87fa508de8799510b49739b65e28335/eras/alonzo/test-suite/src/Test/Cardano/Ledger/Alonzo/Serialisation/Generators.hs#L55-L56